sbc_email/models/import_letters_history.py

```diff
@@ -0,0 +1,438 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014-2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emmanuel Mathier, Loic Hausammann <loic_hausammann@hotmail.com>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+"""
+This module reads a zip file containing scans of mail and finds the relation
+between the database and the mail.
+"""
+import logging
+import base64
+import zipfile
+import tempfile
+import os
+
+from openerp.addons.sbc_compassion.tools import import_letter_functions as func
+from openerp import fields, models, api, _, exceptions
+
+from io import BytesIO
+from openerp.tools.config import config
+from smb.SMBConnection import SMBConnection
+
+from openerp.addons.connector.session import ConnectorSession
+from openerp.addons.sbc_compassion.models import import_letters_history as ilh
+
+logger = logging.getLogger(__name__)
+
+
+class ImportLettersHistory(models.Model):
+    """
+    Keep history of imported letters.
+    This class allows the user to import some letters (individually or in a
+    zip file) in the database by doing an automatic analysis.
+    The code is reading QR codes in order to detect child and partner codes
+    for every letter, using the zxing library for code detection.
+    """
+    _inherit = "import.letters.history"
+
+    from_imports_nas_folder = fields.Boolean(
+        string="Import all content from NAS Imports folder",
+        default=True)
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.multi
+    @api.onchange(
+        "data", "import_line_ids", "letters_ids", "from_imports_nas_folder")
+    def _count_nber_letters(self):
+        """
+        Counts the number of scans. If a zip file is given, the number of
+        scans inside is counted.
+        """
+        for inst in self:
+            if inst.state in ("open", "pending", "ready"):
+                inst.nber_letters = len(inst.import_line_ids)
+            elif inst.state == "done":
+                inst.nber_letters = len(inst.letters_ids)
+            elif inst.state is False or inst.state == "draft":
+                # counter
+                tmp = 0
+                # if files are selected by user
+                if not inst.from_imports_nas_folder:
+                    # loop over all the attachments
```

Comment:
You can remove all the code inside this condition and replace by calling super method.